examples/with-firebase-client-side/credentials/client.js

```diff
@@ -0,0 +1,28 @@
+// TODO: Fill in with your credentials
+const productionCredentials = {
```

Comment:
none of the credentials here are secret?

Reply:
In any case, let's use [Now Secrets](https://zeit.co/docs/v2/serverless-functions/env-and-secrets?query=secrets) here alongside [Next.js environment variables](https://nextjs.org/docs/api-reference/next.config.js/environment-variables) so that the deployment button with ZEIT Now works (it will ask you to fill the secrets before deploying the app).

Reply:
Nope, client credentials don't need to be secret. Great idea for the Deploy button to work though! Will update!!

Reply:
Hey @lfades , I commited some changes. Zeit knows which variables to ask based on the ones in now.json, or I have to configure some other thing?

On another note, I changed the setup just a bit. The app is initialized in a separate file now, and I feel it's **much** cleaner. The only thing that keeps me preocupied is: in `firebase/clientApp.js` I'm checking for typeof window !== 'undefined', because otherwise it won't build. But, if window is undefined (on the server), will the app get initialized ever? I mean, will the code run again once window is defined? I've tested that a couple times and it works perfectly, but I just wanted to make sure!

Reply:
Last q:
.env.build is needed for using secrets on development, but the file is gitignored, obviously. Should I remove it from .gitignore?

Reply:
You shouldn't be using firebase in the server for this example, right? in any case as long as `firebase.initializeApp` doesn't try to do any kind of html  manipulation then it's okay to have it like that

Reply:
Make sure to include steps for adding the environment variables for localhost, for that you can use `dotenv` with a `.env` file and have a `.env.example` that can work as a guide, you can see [this example](https://github.com/zeit/next.js/tree/canary/examples/cms-sanity) for reference.

Reply:
Related to `firebase/clientApp.js`, as long as you don't have any browser errors (like a react mismatch of content) it's okay.

Reply:
> You shouldn't be using firebase in the server for this example, right?

Nope, on the server you need to authenticate with different credentials, and that's out of scope for this example.

> Make sure to include steps for adding the environment variables for localhost...

Will add this!

Reply:
To see if I got it right: We won't use `.env.build` (that's used for `now dev`). Instead, we'll use a regular `.env` and set the variables in next.config.js for them to work on `npm run dev` ?

According to [this](https://nextjs.org/docs/api-reference/next.config.js/environment-variables) we don't need `dotenv`, do we?

Reply:
that configuration is to replace environment variables at build time with the values in there, but `next.config.js` won't load the `.env` file, so that's why `dotenv` is useful there.

Reply:
We don't want to use `now dev` but `next dev` :+1:

Reply:
Do it in the same way that the CMS examples are doing it, like [cms-sanity](https://github.com/zeit/next.js/tree/canary/examples/cms-sanity)

Reply:
Got it! Done